datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -318,9 +433,12 @@ impl Stream for GroupedHashAggregateStream {
                             assert!(!self.input_done);
 
                             if let Some(to_emit) = self.group_ordering.emit_to() {
-                                let batch = extract_ok!(self.emit(to_emit));
+                                let batch = extract_ok!(self.emit(to_emit, false));
                                 self.exec_state = ExecutionState::ProducingOutput(batch);
                             }
+
+                            extract_ok!(self.emit_early_if_necessary());
```

Comment:
```suggestion
                            } else {
                                extract_ok!(self.emit_early_if_necessary());
                            }
```

Reply:
No, `self.group_ordering.emit_to()` may return `None` for all `GroupOrdering::None`, `::Partial`, and `::Full`. So we cannot put it in the else clause...

Reply:
Oh right, there are `None` cases for `Parital` and `Full` too.